File: library/src/test/java/me/proxer/library/api/user/UserInfoEndpointTest.java
package me.proxer.library.api.user;

import me.proxer.library.ProxerTest;
import me.proxer.library.api.ProxerException;
import me.proxer.library.entity.user.UserInfo;
import okhttp3.mockwebserver.MockResponse;
import org.junit.Test;

import java.io.IOException;
import java.util.Date;

import static org.assertj.core.api.Assertions.assertThat;
import static org.assertj.core.api.Assertions.assertThatExceptionOfType;

/**
 * @author Ruben Gees
 */
public class UserInfoEndpointTest extends ProxerTest {

    @Test
    public void testDefault() throws ProxerException, IOException {
        server.enqueue(new MockResponse().setBody(fromResource("user_info.json")));

        final UserInfo result = api.user()
                .info(null, "rubygee")
                .build()
                .execute();

        assertThat(result).isEqualTo(buildTestInfo());
    }

    @Test
    public void testPath() throws ProxerException, IOException, InterruptedException {
        server.enqueue(new MockResponse().setBody(fromResource("user_info.json")));

        api.user().info("123", "rubygee")
                .build()
                .execute();

        assertThat(server.takeRequest().getPath()).isEqualTo("/api/v1/user/userinfo?uid=123&username=rubygee");
    }

    @Test
    public void testUserIdAndUsernameNull() {
        assertThatExceptionOfType(IllegalArgumentException.class)
                .isThrownBy(() -> api.user().info(null, null));
    }

    private UserInfo buildTestInfo() {
        return new UserInfo("121658", "RubyGee", "121658_cEBC8F.png", false, false,
                "Ihr könnt mich jederzeit anschreiben, Skype oder ProxerPn!", new Date(1485372334L * 1000),
                983, 38, 4201, 548, 8, 1700);
    }
}

File: library/src/test/kotlin/me/proxer/library/api/user/UserInfoEndpointTest.kt
package me.proxer.library.api.user

import me.proxer.library.ProxerTest
import me.proxer.library.entity.user.UserInfo
import me.proxer.library.runRequest
import org.amshove.kluent.invoking
import org.amshove.kluent.shouldEqual
import org.amshove.kluent.shouldThrow
import org.junit.jupiter.api.Test
import java.util.Date

/**
 * @author Ruben Gees
 */
class UserInfoEndpointTest : ProxerTest() {

    private val expectedInfo = UserInfo(
        id = "121658", username = "RubyGee", image = "121658_cEBC8F.png", isTeamMember = false, isDonator = false,
        status = "Ihr könnt mich jederzeit anschreiben, Skype oder ProxerPn!",
        lastStatusChange = Date(1485372334L * 1000), uploadPoints = 983, forumPoints = 38, animePoints = 4201,
        mangaPoints = 548, infoPoints = 8, miscPoints = 1700
    )

    @Test
    fun testDefault() {
        val (result, _) = server.runRequest("user_info.json") {
            api.user
                .info(null, "rubygee")
                .build()
                .execute()
        }

        result shouldEqual expectedInfo
    }

    @Test
    fun testPath() {
        val (_, request) = server.runRequest("user_info.json") {
            api.user.info("123", "rubygee")
                .build()
                .execute()
        }

        request.path shouldEqual "/api/v1/user/userinfo?uid=123&username=rubygee"
    }

    @Test
    fun testUserIdAndUsernameNull() {
        invoking { api.user.info(null, null) } shouldThrow IllegalArgumentException::class
    }
}
